lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
```

Comment:
Since the local and remote replicas are separate from each other, is there any reason why the same randomly generated start index couldn't be used for each (choosing the starting local and remote points)?